### DISCUSSION TITLE
Provide support for .cpp files

### DISCUSSION THREAD

**Discussion by BenPag** [bxparks/UnixHostDuino#58] [Ideas]
First one, great work! I like it, you are my hero 👍😃 

Do you want to support `*.cpp` files too? I could need it for my project (currenty I adjust it locally) and would integrate it into your repo, if you are confirm with it.

So I can open a pull request if you want. 

Best greetings ✌️

**Comment by bxparks** [comment]
Thanks!
Not sure what you mean by `*.cpp` files. All my C++ files are named `*.cpp`.

**Reply by BenPag** [reply]
I mean the extension of the test file. This have to be a `.ìno` file. For instance if I rename the `BlinkSOS.ino`  file in the examples folder to `BlinkSOS.cpp` I can not compile it.

**Comment by bxparks** [comment]
Right, that's true, because Arduino expects the main file to be `*.ino`. What's your use case for making it `*.cpp`?

**Reply by BenPag** [reply]
I use PlatformIO and want to test single libraries locally. Because of this I actually do not have `.ino` files in my project. So this feature request is provided by consistency reasons in terms of my project. So technically everything wotks fine if I use `.ino` files.

**Reply by bxparks** [reply]
I don't use PlatformIO, and I don't support it officially. People occasionally send patches for that environment, and I'll accept them if it means no additional time and effort on my part. Maybe you can explain the problem that you have in PlatformIO in more detail, and your proposed solution, so that I can determine if I can support it.

**Reply by BenPag** [reply]
I can try I but it is a bit complex i guess.

For a little bit more context: I have a mono-repo with different projects for different Arduino boards. In this case has each board his own PlatformIO project. Across all projects I have one folder for all shared libraries for intercommunication, shared logic, etc. purpose.

Now my goal is to create a local test environment to run tests for encapsulated libaries and later on integration tests with stubs too.

But in my point of view PlatformIO is not designed for testing on local machines without the specific (Arduino) board. Also it is not possible to test single libraries. I tried to use the native environment of PlatformIO with EpoxyDuino as dependency but this is not working well for me. (For instance I can not switch between stub and real libraries, also the code completion and code linting not working well)

I switched over to using your approch by defining Makefiles for each library and create one parent Makefile to complie and execute all of them at one time. This works all fine, except one bad taste. I have to use `.ino` files for all my tests. And this is now what I nicly want to have :)

I hope you can understand my "problem". Futhermore I will open a PR for you to see the necessary changes.

**Reply by bxparks** [reply]
I guess I'm still not seeing why it's important for the `*.ino` file to be named `*.cpp`. If you are creating Arduino programs, then the Arduino convention is that the `setup()` and `loop()` functions are in the `*.ino` file. It's actually a helpful convention because I know which file to look at to find the entry points of the program. And by using the standard `*.ino` convention, all my unit tests can be compiled and executed on an actual microcontroller board through the Arduino IDE and CLI.

If you change it to a `*.cpp`, the program can no longer be compiled through the Arduino IDE or CLI.

I looked at your PR, and it's minor enough that I'm inclined to accept it. But I still want to understand why you want to break from Arduino conventions if you are programming for Arduino.
